### ISSUE TITLE
To use decap CMS on Github pages. Do I have to host the site on Netlify?

### ISSUE THREAD

**Issue by RDjarbeng** [decaporg/decap-cms#7280]
**Describe the bug**
I followed the jekyll setup instructions and my site still can't use decap CMS.  I followed the instructions here
[jekyll guide](https://decapcms.org/docs/jekyll/) and [here- GitHub backend](https://decapcms.org/docs/github-backend/) to setup the GitHub backend.

I followed the steps to setup the identity widget
I have setup the secrets and got the client ID for my project but at the section that says to:

> Follow the authentication provider setup steps in the [Netlify docs](https://www.netlify.com/docs/authentication-providers/#using-an-authentication-provider)

It says in the netlify docs

> When you complete application registration with GitHub, you need to add the Client ID and Client Secret to your Netlify site:

Problem is that the site isn't hosted on Netlify, since it's on Github pages. I assumed I wouldn't have to change the hosting provider. But the remaining steps require a Netlify site to do these steps in the Netlify UI [page](https://docs.netlify.com/security/secure-access-to-sites/oauth-provider-tokens/#using-an-authentication-provider). Clicking the links takes me to one of my other projects hosted on Netlify so I can enter the client ID and secret.

> Go to Site configuration > Access & security > OAuth
Under Authentication Providers, select Install Provider.
Select GitHub and enter the Client ID and Client Secret from earlier, then save.

So this is where I had to stop following the tutorial on the DecapCMS docs. I feel I'm almost there but just missing something.

I tried to look for alternatives and found the section for external OAuth, but it doesn't have a jekyll setup with Github pages
[https://decapcms.org/docs/external-oauth-clients/](external OAuth)

Nevertheless I tried the serverless option and it just confused me more. If anyone knows how to use this I will be grateful for your help.

**To Reproduce**

**Expected behavior**

To be able to use Decap CMS on my Github pages website without changing hosting providers

**Screenshots**

This is what I see when I go to the /admin route.
![image](https://github.com/user-attachments/assets/eb8140c1-7155-4ac0-870e-a221390e0da9)

Then it fails at this step
![image](https://github.com/user-attachments/assets/c8df5fb7-8367-4ff3-876e-57af0d84e8c3)

**Applicable Versions:**

 - Decap CMS version: 
 - decap-cms-app 3.3.3
decap-cms.js:sourcemap:393 decap-cms-core 3.4.2
decap-cms.js:sourcemap:491 decap-cms 3.3.3
 - Git provider: GitHub
 - OS: Windows 11
 - Browser version Version 128.0.6613.138 (Official Build) (64-bit)

**CMS configuration**

https://github.com/RDjarbeng/RDjarbeng/blob/main/admin/config.yml

**Additional context**

Site is my personal website located at:

https://rdjarbeng.github.io/RDjarbeng/
On GitHub:

https://github.com/RDjarbeng/RDjarbeng

**Comment by RDjarbeng** [comment]
Update (21st September): I've hosted the site on netlify now and managed to get the CMS admin page up and running. 

I still don't like that I have to do this, because it gives me 2 sites to work with; the one on GitHub pages and the one on Netlify. The admin route on the site on github pages is not working. Shows this page with error:
> Unable to access identity settings. When using git-gateway backend make sure to enable Identity service and Git Gateway.

![image](https://github.com/user-attachments/assets/39e5cf8a-a151-44e8-9b1c-75107be5747c)


I read that you can push to netlify and disable builds just so you can use the authentication to sign in. This would mean you only use netlify for the initial setup. But that doesn't work. Once you disable builds you can't push any changes to the collections on the admin console (eg: modifying the config.json) unless you enable the builds.  

So now if I want to continue using decap cms I have to switch between two sites; netlify for the cms, github pages for the actual website instead of just using one site for everything. 

Poor experience using this.😒

**Comment by RDjarbeng** [comment]
Why not just use Netlify?

The site settings are currently set to the options for Github. On netlify the assets folder for some reason isn't found so the site displays without css and other media.

I don't want to change hosting platforms just for a CMS☹️

**Comment by hip3r** [comment]
In short: no, netlify is not a requirement.

As it's described in docs, to use Decap cms it's not necessary to host site on netlify https://decapcms.org/docs/intro/#decap-cms-vs-netlify

The links that you followed, describe that Netlify is used for git authentication (not hosting) https://decapcms.org/docs/jekyll/#enable-authentication-for-cms-users

How to set git gateway without netlify, see https://decapcms.org/docs/git-gateway-backend/#git-gateway-without-netlify
Docs don't have example, since Decap cms allows a lot of custom implementations.
See if this helps https://github.com/decaporg/decap-cms/issues/6370 
If not, try asking a community.

**Comment by RDjarbeng** [comment]
> The links that you followed, describe that Netlify is used for git authentication (not hosting)
Yes, but the steps shown require some site to be set up on netlify. 

I think the docs missed a step where you have to setup some kind of dummy site on netlify (that doesn't necessarily contain the site code), before you can follow the rest of the steps listed here.

' [enable Netlify Identity and Git Gateway services](https://www.decapcms.org/docs/add-to-your-site/#enable-identity-and-git-gateway) for the backend, then [add the Identity widget](https://www.decapcms.org/docs/add-to-your-site/#add-the-netlify-identity-widget) to render a login portal on the frontend.'  -- This won't work if you don't have a site on netlify for your project

>https://decapcms.org/docs/git-gateway-backend/#git-gateway-without-netlify

This link you provided actually only has this as useful information

'You can use [Git Gateway](https://github.com/netlify/git-gateway) without Netlify by setting up your own Git Gateway server and connecting it with your own instance of [GoTrue](https://www.gotrueapi.org/)'

So the title is misleading. Should just be 'git gateway with netlify gotrue' since it only has one option.😒

Thanks anyway

No  #6370 doesn't help.

**Comment by martinjagodic** [comment]
@RDjarbeng you have a point here. I think this is more of a documentation issue, so I'm transferring the issue to that repository. If you have a proposal on how we could improve, please let me know.
